### ISSUE TITLE
FreeBSD support

### ISSUE THREAD

**Issue by centromere** [lpeterse/haskell-socket-sctp#2]
I was able to successfully compile this library on FreeBSD 10.3 by commenting out the `extra-libraries: sctp` line. Is there any way that the cabal file can accommodate FreeBSD?

**Comment by lpeterse** [comment]
Wohoo, this implies the socket library works on FreeBSD :-)

In the socket library I'm doing the following to dispatch between operating systems:

```
  if os(windows)
    include-dirs:      platform/win32/include
    hs-source-dirs:    platform/win32/src
    c-sources:         platform/win32/cbits/hs_socket.c
    extra-libraries:   ws2_32
  else
    include-dirs:      platform/linux/include
    hs-source-dirs:    platform/linux/src
    c-sources:         platform/linux/cbits/hs_socket.c
```

The [cabal documentation](https://www.haskell.org/cabal/users-guide/developing-packages.html#conditions) also mentions `freebsd` as a valid choice, so we could easily achieve that.

Alas, this library is experimental and I have not really used it apart from tinkering around. The `socket` library experienced a breaking change a while ago, because the `0.5.x` branch will stop working when `GHC 8.*` comes out. The `socket-sctp` library still depends on `0.5.x` and should be adapted to `0.6.x` if you plan to do something serious with it.

**Comment by centromere** [comment]
To be fair, I only attempted to compile it. I haven't actually written any code to test the socket library in any meaningful way.

**Comment by shlevy** [comment]
Fixed by #7
